client/js/genre.js

```diff
@@ -28,8 +28,8 @@ async function displayMusicGenre() {
 
   const response = await fetch(`/api/youtube?num_videos=${numVideos}`);
   if (response.status == 401) {
-    // no oauth login so redirect
-    window.location.href = "/api/oauth/login/youtube";
+    // no oauth login so redirect to new page
+    window.open('/api/oauth/login/youtube');
   }
```

Comment:
This redirects to a new page for login which eventually closes. But then the user has to click the button again to login. Do you know if there's any way to wait to proceed with the rest of the function in `displayMusicGenre()` once the new tab closes?

Reply:
It's possible, but tricky. The way that I did it was to have the user redirected to a blank HTML page that [dispatches a `CustomEvent`](https://github.com/laptou/step/blob/598e95e3d5b3515298756a737661323dc88aebf3/portfolio/src/main/java/com/google/sps/servlets/users/LandingServlet.java#L40) to the parent window, then calls `window.close()` to close the child window. The parent window [listens for the `CustomEvent`](https://github.com/laptou/step/blob/598e95e3d5b3515298756a737661323dc88aebf3/portfolio/src/main/client/src/components/controls/authentication.ts#L54), and when it's received, continues executing. In order to understand that last bit of code, you will need to have a good grasp of how Promises work.

Reply:
got it, I'll do that after my MVP